src/PelCanonMakerNotes.php

```diff
@@ -0,0 +1,308 @@
+        0x000d,
+        0x000e,
+        0x000f,
+        0x0010,
+        0x0011,
+        0x0012,
+        0x0013,
+        0x0014,
+        0x0015,
+        0x0017,
+        0x0018,
+        0x0019,
+        0x001b,
+        0x001c
+    );
+
+    private $undefinedFileInfoTags = array(
+        0x0002,
+        0x000a,
+        0x000b,
+        0x0011,
+        0x0012,
+        0x0016,
+        0x0017,
+        0x0018
+    );
+
+    public function __construct($parent, $data, $size, $offset)
+    {
+        parent::__construct($parent, $data, $size, $offset);
+        $this->type = PelIfd::CANON_MAKER_NOTES;
+    }
+
+    public function load()
+    {
+        $this->components = $this->data->getShort($this->offset);
+        $this->offset += 2;
+        Pel::debug('Loading %d components in maker notes.', $this->components);
+        $mkNotesIfd = new PelIfd(PelIfd::CANON_MAKER_NOTES);
+
+        for ($i = 0; $i < $this->components; $i++) {
+            $tag = $this->data->getShort($this->offset + 12 * $i);
+            $type = $this->data->getShort($this->offset + 12 * $i + 2);
+            $components = $this->data->getLong($this->offset + 12 * $i + 4);
+            $data = $this->data->getLong($this->offset + 12 * $i + 8);
+            // check if tag is defined
+            if (in_array($tag, $this->undefinedMakerNotesTags)) {
+                continue;
+            }
+            switch ($tag) {
+                case PelTag::CANON_CAMERA_SETTINGS:
+                    $this->parseCameraSettings($mkNotesIfd, $this->data, $data, $components);
+                    break;
+                case PelTag::CANON_SHOT_INFO:
+                    $this->parseShotInfo($mkNotesIfd, $this->data, $data, $components);
+                    break;
+                case PelTag::CANON_PANORAMA:
+                    $this->parsePanorama($mkNotesIfd, $this->data, $data, $components);
+                    break;
+                case PelTag::CANON_PICTURE_INFO:
+                    $this->parsePictureInfo($mkNotesIfd, $this->data, $data, $components);
+                    break;
+                case PelTag::CANON_FILE_INFO:
+                    $this->parseFileInfo($mkNotesIfd, $this->data, $data, $components);
+                    break;
+                case PelTag::CANON_CUSTOM_FUNCTIONS:
+                    //TODO
+                default:
+                    $mkNotesIfd->loadSingleValue($this->data, $this->offset, $i, $tag);
+                    break;
+            }
+        }
+        $this->parent->addSubIfd($mkNotesIfd);
+    }
+
+    private function parseCameraSettings($parent, $data, $offset, $components)
+    {
+        $type = PelIfd::CANON_CAMERA_SETTINGS;
+        Pel::debug('Found Canon Camera Settings sub IFD at offset %d', $offset);
+        $size = $data->getShort($offset);
+        $offset += 2;
+        $elemSize = PelFormat::getSize(PelFormat::SSHORT);
+        if ($size / $components !== $elemSize) {
+            // TODO throw
```

Comment:
Should an exception be thrown in this case (and in similar other cases?)

Reply:
You're right. Embarrassing I missed the TODOs...
One problem is, that I don't know if it is correct to throw an exception at this point. Since there are unknown tags which could be of another format. Thus it would throw an `PelMalformedMakerNotesException` or something else, just because there is an unknown tag.
What's your opinion? I'm ok with both solutions (1. remove the checks and 2. keep them and adjust if there is a case in the future where the exception is thrown although it is a valid maker notes section).

Reply:
I would throw an exception so we would get bug reports if there is something wrong. Code could be improved. Otherwise strange results could be seen. I haven't worked with those parsers yet, so I don't really knw about the impact.

Reply:
Ok, will add a exception the next days. Do you refer to a specific part for improvement?

Reply:
My idea is that we might see exception reports when users report PelMalformedMakerNotesException .

Reply:
Please have a look at the new scruntizer tests which show 43 new issues and a decrease in code coverage.

Reply:
Most of the issues are not caused by this PR. Code Coverage is difficult. E.g. `PelEntrySShort` got from 100% code coverage to ~2%, because I had to add the `getText` method. It should be possible to increase it a bit, but I added more than 1000 Lines and almost 700 are for different lenses. Thus I'm not sure how much it could be increased and what would be a good goal.

And since scrutinizer was added today I'm not sure if the result are representative in any way. But I'll have a look at the issues/code coverage caused by this PR.